source/hp/dof_handler.cc

```diff
@@ -2675,47 +2689,21 @@ namespace hp
                 ExcInternalError());
       }
 
-    // finally, do the renumbering
-    // and set the number of actually
+    // finally, do the renumbering and set the number of actually
     // used dof indices
-    renumber_dofs_internal (new_dof_indices, dealii::internal::int2type<dim>());
+    policy->renumber_dofs (new_dof_indices);
 
-    // now set the elements of the
-    // number cache appropriately
-    number_cache.n_global_dofs        = next_free_dof;
-    number_cache.n_locally_owned_dofs = number_cache.n_global_dofs;
+    // now set the elements of the number cache appropriately
+    number_cache = dealii::internal::DoFHandler::NumberCache (next_free_dof);
 
-    if (dynamic_cast<const parallel::shared::Triangulation< dim, spacedim >*>
-        (&this->get_triangulation())
-        == nullptr)
-      {
-        number_cache.locally_owned_dofs
-          = IndexSet (number_cache.n_global_dofs);
-        number_cache.locally_owned_dofs.add_range (0,
-                                                   number_cache.n_global_dofs);
-        Assert (number_cache.n_global_dofs < std::numeric_limits<unsigned int>::max (),
-                ExcMessage ("Global number of degrees of freedom is too large."));
-        number_cache.n_locally_owned_dofs_per_processor
-          = std::vector<types::global_dof_index> (1,
-                                                  (types::global_dof_index) number_cache.n_global_dofs);
-      }
-    else
-      {
-        AssertThrow(false, ExcNotImplemented() );
-        //number_cache.locally_owned_dofs = dealii::DoFTools::locally_owned_dofs_with_subdomain(this,tria->locally_owned_subdomain() );
-        //TODO: update n_locally_owned_dofs_per_processor as well
-      }
-
-    number_cache.locally_owned_dofs_per_processor
-      = std::vector<IndexSet> (1,
-                               number_cache.locally_owned_dofs);
-
-    // update the cache used for cell dof indices and compress the data on the levels. do
-    // the latter on separate tasks to gain parallelism, starting with the highest
-    // level (there is most to do there, so start it first)
-    for (active_cell_iterator cell = begin_active();
-         cell != end(); ++cell)
-      cell->update_cell_dof_indices_cache ();
+    Assert ((dynamic_cast<const parallel::shared::Triangulation< dim, spacedim >*>
```

Comment:
Yes, this code is going to go away, and since I'm going to take care of one kind of triangulation at a time, it's easiest to leave both assertions separate.